tests/Resources/ti.ui.window.test.js

```diff
@@ -96,6 +96,146 @@ describe('Titanium.UI.Window', function () {
 		});
 	});
 
+	it.ios('.leftNavButtons and .rightNavButtons', finish => {
+		const rightButton1 = Ti.UI.createButton({
+			title: 'Right1',
+			color: 'green',
+		});
+
+		const rightButton2 = Ti.UI.createButton({
+			title: 'Right2',
+			color: 'green',
+		});
+
+		const leftButton = Ti.UI.createButton({
+			title: 'Left',
+			color: 'blue'
+		});
+
+		const rootWindow = Ti.UI.createWindow({
+			backgroundColor: 'white',
+			leftNavButtons: [ leftButton ],
+			rightNavButtons: [ rightButton1, rightButton2 ],
+		});
+
+		win = Ti.UI.createNavigationWindow({
+			window: rootWindow
+		});
+
+		win.open();
+
+		rootWindow.addEventListener('focus', function focus() {
+			rootWindow.removeEventListener('focus', focus);
+			try {
+				should(rootWindow.rightNavButtons).be.an.Array();
+				should(rootWindow.rightNavButtons.length).be.eql(2);
+				rootWindow.rightNavButtons = [ rightButton1 ];
+				should(rootWindow.rightNavButtons.length).be.eql(1);
+
+				should(rootWindow.leftNavButtons).be.an.Array();
+				should(rootWindow.leftNavButtons.length).be.eql(1);
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+	});
+
+	it.ios('.leftNavButton and .rightNavButton  with color', finish => {
+		this.timeout(10000);
+
+		const nonColorButton = Ti.UI.createButton({
+			title: 'Right',
+		});
+
+		const greenColorButton = Ti.UI.createButton({
+			title: 'Right',
+			color: 'green',
+		});
+
+		const leftButton = Ti.UI.createButton({
+			title: 'Left',
+			color: 'blue'
+		});
+
+		const rootWindow = Ti.UI.createWindow({
+			backgroundColor: 'white',
+			leftNavButton: leftButton,
+			rightNavButton: nonColorButton,
+		});
+
+		win = Ti.UI.createNavigationWindow({
+			window: rootWindow
+		});
+
+		win.open();
+
+		rootWindow.addEventListener('postlayout', function postlayout() {
+			rootWindow.removeEventListener('postlayout', postlayout);
+			setTimeout(function () {
+				const nonColorButtonImage = win.toImage();
+				rootWindow.rightNavButton = greenColorButton;
+				setTimeout(function () {
+					try {
+						should(rootWindow.leftNavButton).be.an.Object();
+						should(rootWindow.rightNavButton).be.an.Object();
+						should(win).not.matchImage(nonColorButtonImage, 0); //  Navbutton without color should be different with a colored one
```

Comment:
The test I had in mind, was to simply change the color for a navbutton and verify the image matches expectations for this PR (namely that the text color is the one you assigned). I suppose it can be useful to verify that one given `color: 'green'` doesn't look the same as one without `color` property specified, but you can get to the same place by simply having one test verify the "default" no-color image, and another verify the green color button. Or simply a single test with multiple buttons: one assigned a color, one assigned a tintColor, one with neither assigned and verify that image.

Reply:
@sgtcoolguy   If you are talking about matching against  hardcoded image, snapshot does not match on iPhone SE (2nd generation) simulator and iPhone 11 simulator even we use 'px' (e.g. 400px) to set width and height of navigation window. Reason looks nav button titles are a bit different in both simulator. 
So I compared screenshots with colored and non colored nav bar button and it should not match.
I had tried  -  
`	it.ios('.leftNavButton and .rightNavButton  with color', finish => {
		const rightButton = Ti.UI.createButton({
			title: 'Right',
			tintColor: 'red',
			color: 'green',
		});

		const leftButton = Ti.UI.createButton({
			title: 'Left',
			color: 'red'
		});

		const rootWindow = Ti.UI.createWindow({
			backgroundColor: 'white',
			leftNavButton: leftButton,
			rightNavButton: rightButton,
		});

		win = Ti.UI.createNavigationWindow({
			height: '400px',
			width: '400px',
			window: rootWindow
		});

		win.open();

		rootWindow.addEventListener('postlayout', function postlayout() {
			rootWindow.removeEventListener('postlayout', postlayout);
			setTimeout(function () {
				try {
					should(rootWindow.leftNavButton).be.an.Object();
					should(rootWindow.rightNavButton).be.an.Object();
					should(win).matchImage('snapshots/navButtonWithColor.png');
				} catch (e) {
					return finish(e);
				}
				finish();
			}, 1000);
		});
	});
`
But screenshots from iPhone 11 Pro max and iPhone SE (2nd generation) simulator didn't match. Am I missing anything?

Reply:
I don't know since I don't have the images to look at manually. I assume that despite the use of pixel dimensions some things may matter based on screen dpi, so you'd have to use the density in the filename of the image and produce copies for varying densities?

Reply:
I tested this locally on this PR, the issue is indeed a scale/density issue whereby the button text is sized differently on different density screens (since text/fonts are always in typographical points on iOS: https://appcelerator.github.io/titanium-docs/api/structs/font.html#fontsize)

Reply:
Yeah. You are right. Different scale images should be added. Added 2x and 3x images. I think no need of scale 1x image as we do not support those devices.